src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,40 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle)obj;
+        if (color == null) {
+            if (rectangle.color != null) {
+                return false;
+            }
+        }
+        if (color == null && rectangle.color == null && width == null && rectangle.width == null
+                && length == null && rectangle.length == null) {
+            return true;
+        }
+
+        if (obj.getClass().equals(Rectangle.class)) {
+            return rectangle.getWidth().equals(this.getWidth()) && rectangle.getLength()
+                    .equals(this.getLength()) && rectangle.getColor().equals(this.getColor());
```

Comment:
let's move null checks here as well (try to write everything in a single statement after you cast the `obj`

also, please do not use `this` keyword when addressing the fields if the current object - it is redundant

Reply:
done